### PR TITLE
fix: role implementation

### PR DESCRIPTION
**Pull request by aisyahnnd**
Role implementation when login as user or admin

### CODE REVIEW

**Comment by aisyahnnd** [comment]
Already fix : cbef370e41314ef02f6d1d206205552a64eeafd3

File: src/App.jsx [hunk 1]
```diff
@@ -18,21 +24,98 @@ const getToken = () => {
...
+  const role = getRole();
+
+  const ProtectedRoute = ({
+    tokenAvaliable,
+    redirectPath = "/login",
+  }) => {
+    const location = useLocation();
+    tokenAvaliable = !!JSON.parse(localStorage.getItem("token"));
+
+    return tokenAvaliable ? (
```
Comment (mazsam) [inline src/App.jsx]: Please fix typo

File: src/App.jsx [hunk 2]
```diff
@@ -18,21 +24,98 @@ const getToken = () => {
   return userToken;
 };
 
+const getRole = () => {
```
Comment (mazsam) [inline src/App.jsx]: Please create a new file, so other can use this function further

File: src/App.jsx [hunk 3]
```diff
@@ -18,21 +24,98 @@ const getToken = () => {
...
+              path="/mytraining/edit/:id"
+              element={<TrainingEditPage />}
+            />
+          </Route>
+        </Route>
         <Route path="*" exact={true} element={<MissingPage />} />
         <Route path="/missing" element={<MissingPage />} />
-        <Route path="/login" element={<LoginPage setToken={setToken} />} />
-        <Route path="/register" element={<RegisterPage />} />
+        <Route element={<ProtectedLogin tokenAvaliable={token} />}>
```
Comment (mazsam) [inline src/App.jsx]: Please fix typo

**Comment by aisyahnnd** [comment]
Already fix : dfc1a5b1259d38ff8a05b082c72c7501945d846b

**Pull request merged**